pkg/controller/pv.go

```diff
@@ -51,33 +56,61 @@ func (r *reconcilePV) Reconcile(ctx context.Context, request reconcile.Request)
 	var pv corev1.PersistentVolume
 	if err := r.client.Get(ctx, request.NamespacedName, &pv); err != nil {
 		if errors.IsNotFound(err) {
-			return reconcile.Result{}, err
+			return reconcile.Result{}, nil
 		}
 		klog.Errorf("failed to get PV (%s): %v", request.Name, err)
 		return reconcile.Result{Requeue: true}, err
 	}
 
-	if pv.Spec.CSI == nil || pv.Spec.CSI.Driver != DriverName {
+	if pv.Spec.CSI == nil || pv.Spec.CSI.Driver != azureutils.DriverName {
 		return reconcile.Result{}, nil
 	}
 
 	var azVolume v1alpha1.AzVolume
 	diskName, err := azureutils.GetDiskNameFromAzureManagedDiskURI(pv.Spec.CSI.VolumeHandle)
+	azVolumeName := strings.ToLower(diskName)
 	if err != nil {
 		klog.Errorf("failed to extract proper diskName from pv(%s)'s volume handle (%s): %v", pv.Name, pv.Spec.CSI.VolumeHandle, err)
 		// if disk name cannot be extracted from volumehandle, there is no point of requeueing
 		return reconcile.Result{}, err
 	}
-	if err := r.client.Get(ctx, types.NamespacedName{Namespace: r.namespace, Name: diskName}, &azVolume); err != nil {
-		// if underlying PV was found but AzVolume was not. Might be due to stale cache
-		klog.V(5).Infof("failed to get AzVolume (%s): %v", diskName, err)
+	if err := r.client.Get(ctx, types.NamespacedName{Namespace: r.namespace, Name: azVolumeName}, &azVolume); err != nil {
+		// if underlying PV was found but AzVolume was not. Might be due to stale cache, so try until found, or maxTry Reached
+		if !errors.IsNotFound(err) {
+			klog.V(5).Infof("failed to get AzVolume (%s): %v", diskName, err)
+			return reconcile.Result{Requeue: true}, err
+		}
+		r.retryMutex.RLock()
+		numRetry, ok := r.retryMap[azVolumeName]
+		r.retryMutex.RUnlock()
+		if !ok {
+			r.retryMutex.Lock()
+			if _, ok = r.retryMap[azVolumeName]; !ok {
+				r.retryMap[azVolumeName] = 1
+				numRetry = 1
+			}
+			r.retryMutex.Unlock()
+		}
+
+		if numRetry <= maxRetry {
+			klog.V(5).Infof("Waiting for AzVolume (%s) to be created...", azVolumeName)
+			r.retryMutex.Lock()
+			r.retryMap[azVolumeName]++
```

Comment:
If the element doesn't exist, the value of retry would become 1 on line 89 and it would become 2 on line 98- even for the first retry. Is this intended or should we use _else if numRetry <= maxRetry_  instead of _if numRetry <= maxRetry_ ?

Reply:
We should be using else if. Will be pushing the change in shortly.

Reply:
On second thought, it should remain as is, given the double-checked locking in place. Instead numRetry should be set to 0 instead of 1 in line 89.